manager-rs/dauth-service/src/manager.rs

```diff
@@ -119,15 +134,96 @@ pub async fn generate_auth_vector(
     Ok(av_response)
 }
 
-/// Finishes the auth process for an auth vector.
-/// Auth vector must have been generated by this network.
-/// Returns the corresponding kseaf value if valid.
+/// Attempts to find the Kseaf value for the network.
+/// Runs the following checks:
+/// 1. If generated on this network, get Kseaf from the database.
+/// 2. Else, check the home network for a Kseaf value.
+/// 3. If not from the home network, get key shares from the backup networks.
 pub async fn confirm_auth_vector(
     context: Arc<DauthContext>,
+    user_id: &str,
     res_star: auth_vector::types::ResStar,
 ) -> Result<auth_vector::types::Kseaf, DauthError> {
     tracing::info!("Confirming auth with res_star: {:?}", res_star);
 
+    let (home_network_id, backup_network_ids) =
+        clients::directory::lookup_user(context.clone(), user_id).await?;
+
+    if home_network_id == context.local_context.id {
+        tracing::info!("User owned by this network");
+        get_confirm_key(context.clone(), res_star).await
+    } else {
+        let (address, _) =
+            clients::directory::lookup_network(context.clone(), &home_network_id).await?;
+
+        let state;
+
+        {
+            let mut map = context.backup_context.auth_states.lock().await;
+            state = map.remove(user_id).ok_or(DauthError::NotFoundError(
+                "Could not find state for auth transaction".to_string(),
+            ))?;
+            // drop mutex guard
+        }
+
+        let xres_star_hash = auth_vector::gen_xres_star_hash(&state.rand, &res_star);
+
+        match state.source {
+            AuthSource::HomeNetwork => {
+                tracing::info!("Auth started from home network");
+                Ok(clients::home_network::get_confirm_key(
+                    context.clone(),
+                    &res_star,
+                    &xres_star_hash,
+                    &address,
+                )
+                .await?)
+            }
+            AuthSource::BackupNetwork => {
+                tracing::info!("Auth started from backup network");
+
+                let mut key_shares = Vec::with_capacity(backup_network_ids.len());
+
+                for backup_network_id in backup_network_ids {
+                    let (backup_address, _) =
+                        clients::directory::lookup_network(context.clone(), &backup_network_id)
+                            .await?;
+                    match clients::backup_network::get_key_share(
+                        context.clone(),
+                        &xres_star_hash,
+                        &res_star,
+                        &backup_address,
+                    )
+                    .await
```

Comment:
This is okay for now, but we definitely will want to make these requests in parallel sooner rather than later! It's totally normal for some subset of the backup networks to be offline, in which case you would be awaiting the rpc timeout, which could be pretty long.

Reply:
Pushed a quick change to address this